### PR TITLE
feature: improve quick stage

### PR DESCRIPTION
**Pull request by yyyc514**
- better status display "x files staged"
    - add "git: commit" to allow quick staging to become quick commiting

### CODE REVIEW

**Comment by yyyc514** [comment]
screen shot 2015-03-27 at 4 06 39 am

**Comment by yyyc514** [comment]
I can fix the pluralization also if you care.

**Comment by divmain** [comment]
@yyyc514, I sent you an email last night.  Can you please check and let me know if you haven't received?

**Comment by yyyc514** [comment]
I just responded, it's not an email address I check all the time.

File: core/commands/quick_stage.py [hunk 1]
```diff
@@ -12,7 +12,8 @@
 CLEAN_WORKING_DIR = "Nothing to commit, working directory clean."
 ADD_ALL_UNSTAGED_FILES = " ?  All unstaged files"
 ADD_ALL_FILES = " +  All files"
-
+STAGED = "--- {} files are staged for commit. ---"
+COMMIT = "    git: commit"
```
Comment (divmain) [inline core/commands/quick_stage.py]: Nit: Add an extra line here.  PEP8 calls for two lines in-between top-level class definitions.

File: core/commands/status_bar.py [hunk 2]
```diff
@@ -6,17 +6,20 @@
 
 class GsStatusBarEventListener(EventListener):
 
-    def on_new(self, view):
+    def trigger(self):
```
Comment (divmain) [inline core/commands/status_bar.py]: This change/method is not necessary.  Please remove.

File: core/commands/quick_stage.py [hunk 3]
```diff
@@ -83,18 +87,29 @@ def get_menu_options(self):
...
                 menu_text = "[{0}] {1}".format(entry.working_status, filename)
                 menu_options.append(MenuOption(True, menu_text, filename, entry.index_status == "?"))
 
-        if not menu_options:
-            return [MenuOption(False, CLEAN_WORKING_DIR, None, None)]
+        if not status_entries:
+            menu_options.append(MenuOption(False, CLEAN_WORKING_DIR, None, None))
+            return menu_options
+
+        if unstaged_count>0:
```
Comment (divmain) [inline core/commands/quick_stage.py]: Spaces around `>`.

File: core/commands/quick_stage.py (hunk 1 above)
Comment (yyyc514) [inline core/commands/quick_stage.py]: I noticed, oversight.  Anything like rubocop for Python?

File: core/commands/quick_stage.py [hunk 4]
```diff
@@ -83,18 +87,29 @@ def get_menu_options(self):
...
                 filename = (entry.path if not entry.index_status == "R"
                             else entry.path + " <- " + entry.path_alt)
                 menu_text = "[{0}] {1}".format(entry.working_status, filename)
                 menu_options.append(MenuOption(True, menu_text, filename, entry.index_status == "?"))
 
-        if not menu_options:
-            return [MenuOption(False, CLEAN_WORKING_DIR, None, None)]
+        if not status_entries:
+            menu_options.append(MenuOption(False, CLEAN_WORKING_DIR, None, None))
+            return menu_options
```
Comment (divmain) [inline core/commands/quick_stage.py]: The change to the `if` line is okay, but change these two lines back to:
```python
return [MenuOption(False, CLEAN_WORKING_DIR, None, None)]
```
And actually, please move this block to the top of the method to return early, immediately after the definition of `status_entries`.

File: core/commands/quick_stage.py [hunk 5]
```diff
@@ -48,6 +49,9 @@ def on_selection(id):
             if not selection.requires_action:
                 return
 
+            elif selection.menu_text == COMMIT:
+                self.window.run_command("gs_commit")
+
```
Comment (divmain) [inline core/commands/quick_stage.py]: Add line with `return` here.

File: core/commands/quick_stage.py (hunk 1 above)
Comment (divmain) [inline core/commands/quick_stage.py]: I use SublimeLinter and the [PEP8](https://github.com/SublimeLinter/SublimeLinter-pep8) plugin.  I think you can also just `pip3 install pep8` and then run `pep8 FILENAME` from the terminal.

**Comment by divmain** [comment]
I like this change!  Comments are pretty minor, once they're done I'll merge.

**Comment by yyyc514** [comment]
> I like this change!

I know. See... I'm not as crazy as you think. :-)

File: core/commands/quick_stage.py (hunk 1 above)
Comment (yyyc514) [inline core/commands/quick_stage.py]: Do you have a list of which errors you care about vs ignore so I can configure this a little tighter?  Right now I'm getting flags for lines longer than 79, which seems to be  pretty common issue in this source. :)

**Pull request merged**